src/main/java/gel/TaskList.java

```diff
@@ -0,0 +1,105 @@
+package gel;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import gel.exception.GelException;
+import gel.task.Deadline;
+import gel.task.Event;
+import gel.task.Task;
+import gel.task.Todo;
+
+/**
+ * Contains the task list e.g., it has operations to add/delete tasks in the list.
+ */
+public class TaskList {
+    private List<Task> listOfTasks;
+    private Ui ui;
```

Comment:
Should `TaskList` contain `Ui`? Your `Parser` class contains `Ui` as well. So you could invoke the methods for `TaskList` and `Ui` from `Parser` instead of invoking some of `Ui` methods in this `TaskList` class. 
An example from your `Parser` class would be this:
```Java
case "bye": { //bye
    storage.updateFile(taskList);
    ui.farewellMessage();
    break label;
}
```
This reduces the dependency between your `TaskList` and `Ui` class and allows your code to be easier to maintain.